### PR TITLE
Add package set commit message

### PR DESCRIPTION
**Pull request by colinwahl**
Adds a commit message to package set publishing of the form:

```
Release 2.0.0 package set.

New packages:
  - math@1.0.0

Updated packages:
  - effect@4.0.0 -> 4.0.1

Removed packages:
  - assert@6.0.0
```

### CODE REVIEW

**Comment by colinwahl** [comment]
Ah, I just realized this will include the headings even when there were no changes of that type. Let me fix that.

**Comment by thomashoneyman** [comment]
The tests are failing because we're now using the `Version` type everywhere and are parsing versions like `"6.0.0"`, but should be parsing (in lenient mode) a version like `"v6.0.0"`.

File: ci/test/Registry/PackageSet.purs [hunk 1]
```diff
@@ -158,17 +199,17 @@ legacyPackageSetDhall =
     }
 }"""
 
-assert :: Tuple String String
-assert = Tuple "assert" "6.0.0"
+assert :: Tuple PackageName Version
+assert = Tuple (unsafeName "assert") (unsafeVersion "6.0.0")
```
Comment (thomashoneyman) [inline ci/test/Registry/PackageSet.purs]:
```suggestion
assert :: Tuple PackageName Version
assert = Tuple (unsafeName "assert") (unsafeVersion "v6.0.0")
```

File: ci/test/Registry/PackageSet.purs [hunk 2]
```diff
@@ -158,17 +199,17 @@ legacyPackageSetDhall =
...
 
-console :: Tuple String String
-console = Tuple "console" "6.0.0"
+console :: Tuple PackageName Version
+console = Tuple (unsafeName "console") (unsafeVersion "6.0.0")
```
Comment (thomashoneyman) [inline ci/test/Registry/PackageSet.purs]:
```suggestion
console :: Tuple PackageName Version
console = Tuple (unsafeName "console") (unsafeVersion "v6.0.0")
```

File: ci/test/Registry/PackageSet.purs [hunk 3]
```diff
@@ -158,17 +199,17 @@ legacyPackageSetDhall =
...
 
-effect :: Tuple String String
-effect = Tuple "effect" "4.0.0"
+effect :: Tuple PackageName Version
+effect = Tuple (unsafeName "effect") (unsafeVersion "4.0.0")
```
Comment (thomashoneyman) [inline ci/test/Registry/PackageSet.purs]:
```suggestion
effect :: Tuple PackageName Version
effect = Tuple (unsafeName "effect") (unsafeVersion "v4.0.0")
```

File: ci/test/Registry/PackageSet.purs [hunk 4]
```diff
@@ -158,17 +199,17 @@ legacyPackageSetDhall =
...
 
-prelude :: Tuple String String
-prelude = Tuple "prelude" "6.0.0"
+prelude :: Tuple PackageName Version
+prelude = Tuple (unsafeName "prelude") (unsafeVersion "6.0.0")
```
Comment (thomashoneyman) [inline ci/test/Registry/PackageSet.purs]:
```suggestion
prelude :: Tuple PackageName Version
prelude = Tuple (unsafeName "prelude") (unsafeVersion "v6.0.0")
```

File: ci/test/Registry/PackageSet.purs [hunk 5]
```diff
@@ -42,12 +43,52 @@ spec = do
...
+    let
+      operations = Map.fromFoldable
+        [ map (const Nothing) assert
+        , Tuple (unsafeName "math") (Just (unsafeVersion "1.0.0"))
+        ]
+
+    PackageSet.commitMessage packageSet operations `Assert.shouldEqual` packageSetCommitMessageNoUpdates
+
+packageSetCommitMessage :: String
+packageSetCommitMessage =
```
Comment (thomashoneyman) [inline ci/test/Registry/PackageSet.purs:66]: I think this should still include the message about the version of the package set that was released. For example, see the existing:
https://github.com/purescript/registry-preview/commit/8f3c33704d8d9601479c2a8d44bc2618ac16ca7c

So the message would be something like:

```
Release 1.0.0 package set

New packages:
  - math@1.0.0
```

**Review by thomashoneyman — CHANGES_REQUESTED** [review]
Just need to make one change wrt the contents of the commit message and this otherwise looks great!

File: ci/test/Registry/PackageSet.purs (hunk 5 above)
Comment (colinwahl) [reply ci/test/Registry/PackageSet.purs:66]: Yes, good catch!

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**